### DISCUSSION TITLE
Manually set person's thumb, but reset during marker

### DISCUSSION THREAD

**Discussion by chain710** [photoprism/photoprism#2480] [Q&A]
Manually set subject(person)'s thumb:

```mysql
update subjects set thumb= "aaa-bbb" where subj_uid="xxx";
```

And calling marker api(marker/$marker_uid undo the work

I did some digging: in UpdateSubjectCovers
```
UPDATE ? LEFT JOIN (
    	SELECT m.subj_uid, m.q, MAX(m.thumb) AS marker_thumb FROM ? m
			WHERE m.subj_uid <> '' AND m.subj_uid IS NOT NULL
			  AND m.marker_invalid = 0 AND m.thumb IS NOT NULL AND m.thumb <> ''
			GROUP BY m.subj_uid, m.q
			) b ON b.subj_uid = subjects.subj_uid
		SET thumb = marker_thumb WHERE ?
```

It seems every time UpdateSubjectCovers is called, all subjects thumb are reset

Questions:
1. In the sql: maybe it's better select max(m.q), and group by only m.subj_uid?
I assume m.q means face quality, and max(m.q) should mean most relavant?
And "group by m.q" get multi records per subj_uid, but here we only need one

2. How can I pin the thumb(don't change with incoming new markers) for certain person
Is using the thumb_src a good idea? Update thumb_src along with thumb, so that where clause should filter those subjects
Or add new field like pinned_thumb?

**Comment by lastzero** [comment]
`thumb_src = 'manual'` would mark the thumb as manually selected, so it should not be updated automatically anymore.

**Reply by chain710** [reply]
Thank you for the reply! 
How about question 1, why use max(thumb)?

**Reply by lastzero** [reply]
Since we only need one thumbnail, this seemed to make sense at the time of implementation. Feel free to send a pull request with an optimized query if you think it is bad and can be improved. Note, however, that the query must work on all supported databases, including SQLite and MaridaDB 10.5.

**Reply by lastzero** [reply]
For example, one reason could be that the query is deterministic. There are other ways to limit the result, but they may not be deterministic, which means that the preview image would change constantly.

**Comment by graciousgrey** [comment]
You can find more information on how to manually set covers for people in the comments here: https://github.com/photoprism/photoprism/issues/383 It's important to set thumb_src to manual so that your changes won't get overwritten.

**Reply by Artekbed** [reply]
Please tell me how to set thumb_src to manual? I can't find anywhere.
